src/gui/guiChatConsole.cpp

```diff
@@ -773,6 +798,52 @@ bool GUIChatConsole::weblinkClick(s32 col, s32 row)
 	return false;
 }
 
+#ifdef USE_CURL
+// This is a function for CURL to write the page's html into a string
+size_t writeHtml(void* contents, size_t size, size_t nmemb, std::string* strptr) {
+    size_t totalSize = size * nmemb;
+    strptr->append((char*)contents, totalSize);
+    return totalSize;
+}
+#endif
+
+void GUIChatConsole::weblinkClickTitle(s32 col, s32 row){
+	std::ostringstream msg;
+	msg << " * ";
+#ifdef USE_CURL
+	std::string weblink, html, title;
+	weblink = getLinkAt(col, row);
+	CURL* curl = curl_easy_init();
+
+	if(weblink.empty() || !curl){
+		return;
+	}
+
+	curl_easy_setopt(curl, CURLOPT_URL, weblink.c_str());
+	curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writeHtml);
+	curl_easy_setopt(curl, CURLOPT_WRITEDATA, &html);
+	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L); // Important for services like https://bit.ly or https://is.gd
+	curl_easy_setopt(curl, CURLOPT_TIMEOUT, 10L); // As we don't know if we are stuck in an infinite loop of redirects or if the specifiec server exists at ALL, 10 secs is max!
+	CURLcode res = curl_easy_perform(curl);
+	curl_easy_cleanup(curl);
+
+	if(res == CURLE_OPERATION_TIMEDOUT || res == CURLE_COULDNT_RESOLVE_HOST) {
+		msg  << gettext("Host does not exist or timeout for 3rd party webpages reached!");
+	} else {
+		if (res != CURLE_OK || html.find("<title>") == std::string::npos || html.find("</title>") == std::string::npos) {
+			msg << gettext("Unkown error while fetching page title!");
+		}
+		size_t start = html.find("<title>") + 7; // +7 so the html tag isn't in there...
+		size_t end = html.find("</title>");
+		title = html.substr(start, end - start);
```

Comment:
this doesn't work for:
```
<title what="ever">The title here</title>

or

<title>The title here</ title>
```

it also won't correctly decode things such as `&quot;`

Reply:
So I should check if the title start tag is before the end tag in the string?

Reply:
That has nothing to do with the problem

Reply:
I _think_ my last commit fixed what you meant, the code itself isn't that clean, yet I am not too sure what an alternative way of implementing something like that would be.

Note: I did not fix </ title>, as it's not valid HTML (the entire page's source following that tag would be in the title)

Reply:
`end >= start` isn't checked either: A malicious website could use this to trigger undefined behavior, since `end - start` will be `< 0` in that case and casting it to an unsigned type will be UB, typically yielding a very large unsigned value, most likely triggering a crash.